### ISSUE TITLE
Datepicker: enhance accessibility.

### ISSUE THREAD

**Issue by chriscorwin** [ExactTarget/fuelux#1431]
The datepicker control is woefully unaccessible.

Even ignoring being usable by persons with disabilities, it ought to be enhanced to be able to be completely keyboard-driven.

Currently, once the datepicker control has `::focus` , the keyboard is completely useless, and one _must_ mouse to pick a date.

Tabbing onto the control should launch the datepicker control.

Tabbing again should close the datepicker control and apply focus to the next element in the tab order.

When the datepicker control is open, arrow keys should move focus between selectable elements within the control, following logical tab index order, but starting with an _optionally settable default_.

The default should be able to be set for either the month/year to be the initial focus or the "days" to be the initial focus, depending upon the intent of the date in question. For instance, if the goal of the form input is to select a birthday, it is likely the user will be wishing to pick a year first. If the goal is to pick a delivery date for an upcoming email message, it is likely the user will be selecting a date first.

Typing numbers should jump to visible selectable dates, and visual affordance should be given that the `::focus` has changed.

With the month/year controls showing typing a number should jump the years control to the typed number, as it matches, starting on the _second_ number. For instance, if the user types a `1` do nothing until the user also then types a `9`, at which point jump to the latest year in the dataset that begins with "19" (this is very likely to be 1989, of course).

A timeout should be set so that if the user abandons typing numbers, and then starts typing numbers _again_, it should count as a _new_ entry (they likely intended to start back at `1`, not type `11` if there were several seconds between inputs).

I recommend informal user testing to figure out exactly how long this timeout should be, but I believe 2 seconds is a good starting point.

Starting to type a month name should jump to the month typed, it can wait until a confirmed match, like on the first three letters or on common abbreviations. Thought for internationalization should be done, too. I recommend using Twitter's [Bloodhound](https://github.com/twitter/typeahead.js/blob/master/doc/bloodhound.md) engine for pattern matching.

## Keyboard Control

*Up:* When a day has `::focus`, move the focus to the date exactly one week prior. When a month has focus, move back one month. When year has focus, move back one year.

*Down:* When a day has `::focus`, move the focus to the date exactly one week later. When a month has focus, move forward one month. When year has focus, move forward one year.

*Left:* When a day has `::focus`, move the focus to the date exactly one day prior. When a month has focus, move back one month. When year has focus, move back one year.

*Right:* When a day has `::focus`, move the focus to the date exactly one day later. When a month has focus, move forward one month. When year has focus, move forward one year.

*Page up/down (and option/control arrow up/down):*  jump between months, in three week intervals, so that the last-displayed week is becomes the _first-displayed_ week. Do not simply jump by "month": such a large jump will leave the user feeling disoriented.

*Option/Control + Page up/down (and option/control arrow left/right):* jump between years, jumping to the exact view (where possible) in the previous or next year, as appropriate.

*Enter:* When a day has `::focus`, populate the date `input` and close the datepicker control, leaving the focus on the date `input`. When the month or year have focus, jump to the selected month/year combo and show the "calendar" view again, moving the focus to the previously-focused date. (enter again at this stage would populate the `input` and close the control.

*Spacebar:* toggle the state of the datepicker control, apply focus as applicable, either to the date `input` or to the default date _or_ to the last-selected date if the user has previously interacted with the datepicker control in question.

*Tab away:* populate the date `input` and close the datepicker control, applying focus on the next form element.

*Escape:* make no changes to the date `input` and close the datepicker control, leaving focus on the date `input`.

**Comment by chriscorwin** [comment]
Research should be done on providing voiceover type support, such as...

- "Calendar is showing, pick a date."
- "Month and Years control is showing, select a month and a year."
- "March 1988 was selected, calendar controls is now showing. Select a day to populate the Birthday input and close the datepicker control."
- "Jumping to March 1974"
- "Jumping ahead three weeks."

?

**Comment by futuremint** [comment]
I would recommend using SLDS for your datepicker now.  Its far more accessible ;)
